frontend/src/component/Account/UserSpacesConfig.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { translate } from 'react-i18next'
+import { BtnSwitch, IconButton, ConfirmPopup, ROLE_LIST } from 'tracim_frontend_lib'
+import { newFlashMessage } from '../../action-creator.sync.js'
+import { deleteWorkspaceMember } from '../../action-creator.async.js'
+
+export class UserSpacesConfig extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = { spaceBeingDeleted: null }
+  }
+
+  handleConfirmDeleteSpace = async () => {
+    const { props } = this
+    const spaceId = this.state.spaceBeingDeleted
+    if (!spaceId) return
+    this.setState({ spaceBeingDeleted: null })
+
+    const fetchResult = await props.dispatch(deleteWorkspaceMember(spaceId, props.userToEditId))
+    if (fetchResult.status !== 204) {
+      props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
+    }
+  }
+
+  render = () => {
+    const { props } = this
+
+    const entries = props.workspaceList.map(space => {
+      if (space.memberList.length > 0) {
+        const member = space.memberList.find(u => u.id === props.userToEditId)
+        if (!member) return
+        const memberRole = ROLE_LIST.find(r => r.slug === member.role)
+        return (
+          <tr key={space.id}>
+            <td>
+              <div className='spaceconfig__table__spacename'>
+                {space.label}
+              </div>
+            </td>
+
+            <td>
+              <div className='spaceconfig__table__role'>
+                <div className='spaceconfig__table__role__icon'>
+                  <i className={`fa fa-fw fa-${memberRole.faIcon}`} style={{ color: memberRole.hexcolor }} />
+                </div>
+                <div className='spaceconfig__table__role__text d-none d-sm-flex'>
+                  {props.t(memberRole.label)}
+                </div>
+              </div>
+            </td>
+
+            <td>
+              <BtnSwitch
+                checked={member.doNotify}
+                onChange={() => props.onChangeSubscriptionNotif(space.id, !member.doNotify)}
+              />
+            </td>
+            <td>
+              <IconButton
+                className='outlineTextBtn primaryColorBorder primaryColorBgHover primaryColorBorderDarkenHover'
+                onClick={() => this.setState({ spaceBeingDeleted: space.id })}
+                icon='sign-out'
+                text={props.admin ? props.t('Remove from space') : props.t('Leave space')}
+              />
+            </td>
+          </tr>
+        )
+      }
+    }).filter(entry => entry)
+
+    return (
+      <div className='account__userpreference__setting__spacename'>
+        <div className='spaceconfig__sectiontitle subTitle ml-2 ml-sm-0'>
+          {props.t('Spaces')}
+        </div>
+
+        <div className='spaceconfig__text ml-2 ml-sm-0' />
+
+        {(entries.length ? (
```

Comment:
we usually indent the ? and : at a new line

Reply:
Should we add an eslint rule for this, to avoid having to check this during review?

Reply:
standardjs already state that rule
If it doesn't show a warning in this case is more likely to be a standardjs issue
So no, we won't add an eslint rule for that case

Reply:
I don't think so.

Reply:
https://standardjs.com/rules.html
look for "ternary operator"

Reply:
Ah right, this is strange that standard did not complain then